core/src/main/scala/zio/config/ConfigDescriptor.scala

```diff
@@ -32,6 +32,8 @@ sealed trait ConfigDescriptor[K, V, A] { self =>
   final def <+>[B](that: => ConfigDescriptor[K, V, B]): ConfigDescriptor[K, V, Either[A, B]] =
     self orElseEither that
 
+  final def at(k: K): ConfigDescriptor[K, V, A] = ConfigDescriptor.nested(k)(this)
```

Comment:
I would delete this operator as it will no longer be necessary after other changes I proposed.

Reply:
Done. Though I'd prefer to keep this method: `nested(path)(inner)` looks too verbose for me.